swap/peer.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2019 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package swap
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	cswap "github.com/ethersphere/swarm/contracts/swap"
+	"github.com/ethersphere/swarm/log"
+	"github.com/ethersphere/swarm/p2p/protocols"
+)
+
+var ErrSigDoesNotMatch = errors.New("signature does not match")
+var ErrDontOwe = errors.New("no negative balance")
+
+// Peer is a devp2p peer for the Swap protocol
+type Peer struct {
+	*protocols.Peer
+	swap        *Swap
+	backend     cswap.Backend
+	beneficiary common.Address
+}
+
+func NewPeer(p *protocols.Peer, s *Swap, backend cswap.Backend, beneficiary common.Address) *Peer {
+	return &Peer{
+		Peer:        p,
+		swap:        s,
+		backend:     backend,
+		beneficiary: beneficiary,
+	}
+}
+
+// handleMsg is for handling messages when receiving messages
+func (sp *Peer) handleMsg(ctx context.Context, msg interface{}) error {
+	switch msg := msg.(type) {
+
+	case *EmitChequeMsg:
+		//return sp.handleEmitChequeMsg(ctx, msg)
+		return sp.handleEmitChequeMsg(ctx, msg)
+
+	case *ErrorMsg:
+		return sp.handleErrorMsg(ctx, msg)
+
+	case *ConfirmMsg:
+		return sp.handleConfirmMsg(ctx, msg)
+	}
+
+	return nil
+}
+
+// handleEmitChequeMsg should be handled by the creditor when it receives
+// a cheque from a creditor
+// TODO: validate the contract address in the cheque to match the address given at handshake
+// TODO: this should not be blocking
+func (sp *Peer) handleEmitChequeMsg(ctx context.Context, msg interface{}) error {
+	log.Info("received emit cheque message")
```

Comment:
This is not a very useful message - maybe add `peer` field, so that at least we know who our node is talking to (we can cross-ref that to the Kademlia table if necessary). Also probably should be `log.Debug`.

Reply:
for simulations it is best if you also log self address

Reply:
I've applied changes based on @nonsense's comment.

I have not added @zelig's suggestion, since I would like this to be consistent and be implemented in _all_ the relevant log calls.

For this, I've created issue #1623.